jaeger-jaxrs2/src/main/java/com/uber/jaeger/filters/jaxrs2/ServerFilter.java

```diff
@@ -94,10 +94,10 @@ public void filter(
         // hitting this case means previous filter was not called
         return;
       }
-      serverSpan = traceContext.pop();
+      serverSpan = traceContext.getCurrentSpan();
 
       Tags.HTTP_STATUS.set(serverSpan, containerResponseContext.getStatus());
-      serverSpan.finish();
+      traceContext.pop();
```

Comment:
please add comments explaining this order (i.e. that we cannot call `pop` as before since it will deactivate & finish the span).

Reply:
is this the only place in the whole project where we call `pop`?

Reply:
Will do. No, we call it in our wrapped Callable and Runnable